core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionIntegrationSuite.scala

```diff
@@ -266,18 +266,17 @@ class BlockManagerDecommissionIntegrationSuite extends SparkFunSuite with LocalS
     val execIdToBlocksMapping = storageStatus.map(
       status => (status.blockManagerId.executorId, status.blocks)).toMap
     // No cached blocks should be present on executor which was decommissioned
-    assert(execIdToBlocksMapping(execToDecommission).keys.filter(_.isRDD).toSeq === Seq(),
+    assert(
+      !execIdToBlocksMapping.contains(execToDecommission) ||
+      execIdToBlocksMapping(execToDecommission).keys.filter(_.isRDD).toSeq === Seq(),
       "Cache blocks should be migrated")
     if (persist) {
       // There should still be all the RDD blocks cached
       assert(execIdToBlocksMapping.values.flatMap(_.keys).count(_.isRDD) === numParts)
     }
 
-    // Make the executor we decommissioned exit
-    sched.client.killExecutors(List(execToDecommission))
-
-    // Wait for the executor to be removed
-    executorRemovedSem.acquire(1)
+    // Wait for the executor to be removed automatically after migration.
+    assert(executorRemovedSem.tryAcquire(1, 5L, TimeUnit.MINUTES))
```

Comment:
So this is a high timeout because github actions seems to occasionally be very slow and I figured a higher value here was safer from a test reliability perpsective. I've normally seen these complete in under a minute on my machine. I think it's better to have the high timeout here given our multiple CI environments.

Reply:
I get that it can take up to 5 minutes, but I would like to understand better whether it should take 5 minutes ? The executor is supposed to exit as soon as the migration is over, right ? Or if there is some polling there -- that polling should ideally be configurable. So the only thing we are really waiting for here is the migration : That migration shouldn't take 5 minutes -- we don't have that much data to migrate.

Perhaps you can greatly reduce the executor heartbeat interval such  that this can be sped up ? (I use this trick in the other block manager decommissioning test)

Reply:
It doesn't take up to 5 minutes in my experience, I'm just picking a high enough timeout it won't flake. I'm not open to changing this, I've dealt with too many flaky tests from timeouts that were set too low.

Reply:
Sounds good. Would it be okay to add a comment about how long does it usually take in the code itself and mention that the larger timeout is for good measure ?